### PR TITLE
New Container image added - Pegasus

### PR DESCRIPTION
**Pull request by dafrenchyman**
This is a pretty big PR adding a new docker container for the Pegasus frontend.

- Controllers should work out of the box for the emulators that can't be setup from within their respective apps. This has been testing via Moonlight running on an XBOX and on Chromebooks.
  - PCSX2
  - RPCS3
  - YUZU
  - ...
- Preliminary CEMU support. Controller and Audio still don't work correctly.

### CODE REVIEW

File: images/es-de/setup.sh
Comment (ABeltramo) [inline images/es-de/setup.sh:1]: I think this shouldn't be part of the commit.
I have similar scripts in my home folder, maybe we should make a general `build.sh` that takes a couple of args and does this? I'm open to feedback and ideas here.

File: images/nvidia-driver/setup.sh
Comment (ABeltramo) [inline images/nvidia-driver/setup.sh:1]: Same as the other `setup.sh` scripts

File: images/pegasus/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+ARG DEBIAN_FRONTEND=noninteractive
+# see: https://github.com/AppImage/AppImageKit/wiki/FUSE#docker
+ENV APPIMAGE_EXTRACT_AND_RUN=1
+
+# Install prereqs
+RUN \
+    echo "**** Install Prereqs (Mesa/Vulkan/Fuse/QT/Misc) ****" && \
+		apt-get update && \
+		apt-get install -y software-properties-common gpg-agent wget p7zip-full libvulkan1 mesa-vulkan-drivers libglu1-mesa-dev qtbase5-dev qt5-qmake ffmpeg x11-xserver-utils libdbus-1-3 libgtk-3-0 libegl1 libsdl2-2.0-0 && \
```
Comment (ABeltramo) [inline images/pegasus/Dockerfile:14]: `mesa-vulkan-drivers` and `libvulkan1` should already be in `base-app`. It's probably best to remove those two here.

File: images/pegasus/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	# Cleanup \
+		apt-get autoremove -y && \
+		rm -rf /var/lib/apt/lists/*
+
+# Install RetroArch
+RUN \
+    echo "**** Install RetroArch ****" && \
+		apt-get update && \
+		apt-get install -y --no-install-recommends software-properties-common gpg-agent libusb-1.0-0 libglu1-mesa libaio1 libaio-dev && \
```
Comment (ABeltramo) [inline images/pegasus/Dockerfile:23]: Could this be `from: ghcr.io/games-on-whales/retroarch` instead?

File: images/pegasus/README.md
Comment (ABeltramo) [inline images/pegasus/README.md:1]: I know all other images don't have docs but it would be good to write down a few lines, like:
 - original project (with link)
 - things included in the image

File: images/pegasus/configs/app/metadata.pegasus.txt
Comment (ABeltramo) [inline images/pegasus/configs/app/metadata.pegasus.txt:1]: Does this stuff really needs to be hardcoded in our repo or can we download it from somewhere (maybe the Pegasus project)?

File: images/pegasus/configs/dolphin/GCPadNew.ini [hunk 3]
```diff
@@ -0,0 +1,120 @@
+[GCPad1]
+Device = evdev/0/Wolf X-Box One (virtual) pad
```
Comment (ABeltramo) [inline images/pegasus/configs/dolphin/GCPadNew.ini:2]: This needs to be better fleshed out; Wolf next release will support hotplug and different types of joypads to better match the client configuration (Xbox/PS/Switch). Why do we need to hardcode the device type like that?

File: images/pegasus/setup.sh
Comment (ABeltramo) [inline images/pegasus/setup.sh:1]: Same as the other `setup.sh` scripts

**Review by ABeltramo — CHANGES_REQUESTED** [review]
Thank you so much for sharing this PR!
I've not (yet) built and tested the image locally, but I've left a few comments over the files that you pushed. Let me know what you think!

File: images/pegasus/configs/dolphin/GCPadNew.ini (hunk 3 above)
Comment (dafrenchyman) [reply images/pegasus/configs/dolphin/GCPadNew.ini:2]: Question about this: I was under the impression that wolf/moonlight just mapped everything to xbox controllers, is that not the case? If this isn't a good idea, would making a script that someone can run to setup different controllers be be better?

File: images/pegasus/configs/app/metadata.pegasus.txt
Comment (dafrenchyman) [reply images/pegasus/configs/app/metadata.pegasus.txt:1]: I wrote this actually. It does need to be hard coded for the bash scripts I wrote. This is very similar to what you did with: https://github.com/games-on-whales/gow/tree/master/images/es-de/scripts/custom_scripts . These our the `custom_scripts` for pegasus.

File: images/pegasus/README.md
Comment (dafrenchyman) [reply images/pegasus/README.md:1]: I can write something up and make it cleaner.

File: images/pegasus/Dockerfile (hunk 2 above)
Comment (dafrenchyman) [reply images/pegasus/Dockerfile:23]: Not a bad idea. Could we also do the same for the `es-ds` image too?

File: images/pegasus/Dockerfile (hunk 1 above)
Comment (dafrenchyman) [reply images/pegasus/Dockerfile:14]: I can remove and test.

File: images/es-de/setup.sh
Comment (dafrenchyman) [reply images/es-de/setup.sh:1]: I think it would be a very good idea to have some kind of build script. Even an `nvidia-build.sh` script. That would make the documentation much simpler too.

File: images/pegasus/Dockerfile (hunk 1 above)
Comment (dafrenchyman) [reply images/pegasus/Dockerfile:14]: I just looked through the code...`mesa-vulkan-drivers` is available, but `libvulkan1` is not. I'm fixing this in `es-de` as well.

Comment (dafrenchyman) [reply images/pegasus/Dockerfile:14]: I won't mess with the base app.

**Comment by drbaumgartne** [comment]
Wasn't quite sure how to submit changes to an existing PR.  A few errors when I tried to build this.
Retroarch ppa needs to move to testing for the Ubuntu-Noble release in the Dockerfile.
Startup.sh is missing the YUZU2 config variable
Startup.sh is has some cp command errors in the CEMU section and RPCS3 sections.
And it would be nice to have a pegasus.yml for the apps folder.

**Comment by dafrenchyman** [comment]
> Wasn't quite sure how to submit changes to an existing PR. A few errors when I tried to build this. Retroarch ppa needs to move to testing for the Ubuntu-Noble release in the Dockerfile. Startup.sh is missing the YUZU_CFG_DIR2 config variable Startup.sh is has some cp command errors in the CEMU section and RPCS3 sections. And it would be nice to have a pegasus.yml for the apps folder.

I will fix

**Comment by dafrenchyman** [comment]
There is another PR now though

**Comment by dafrenchyman** [comment]
I just fixed the above errors as well
